packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -242,6 +247,25 @@ export abstract class AbstractMessageManager<
     this.setMessageStatus(messageId, 'signed');
   }
 
+  /**
+   * Sets a EncryptionPublicKey status to 'received' via a call to this.setMsgStatus and
+   * updates that EncryptionPublicKey in this.messages by adding the raw data of request
+   * to the EncryptionPublicKey.
+   *
+   * @param messageId - The id of the Message to sign.
+   * @param rawSig - The encryption public key of the request.
+   */
+  setMessageStatusReceived(messageId: string, rawSig: string) {
+    const message = this.getMessage(messageId);
+    /* istanbul ignore if */
+    if (!message) {
+      return;
+    }
+    message.rawSig = rawSig;
```

Comment:
I think this is a nice compromise for now as it's only used internally, and we could add a nice generic `result` or something in future, maybe even with a template to have a dynamic type.

Reply:
The content is the same as `setMessageStatusSigned` except that it sets the status to `received` instead of `signed`.

I've created a new method `setMessageStatusAndResult` that includes a new `status` parameter and I've refactored `setMessageStatusSigned` to use it to keep backwards compatibility.

Then we can call `setMessageStatusAndResult` directly from the controller.